src/app_model/backends/qt/_qaction.py

```diff
@@ -146,6 +148,29 @@ def __init__(
         with contextlib.suppress(NameError):
             self.update_from_context(self._app.context)
 
+    def _on_triggered(self, checked: bool) -> None:
+        # Pass bound `QModelMenu` objects when callback parameter annotation matches
+        callback = self._app.commands[self._command_id].resolved_callback
+        sig = _resolve_sig_or_inform(
```

Comment:
this really isn't something that should be used at callback time, it's expensive and complicated.  the goal of this function in in-n-out was to make sure, *at injection time* that we had everything we needed to later match up the types to providers.  But I don't like the idea of resolution being done every time a command is called.